src/algebra/direct_sum_graded.lean

```diff
@@ -433,6 +455,130 @@ instance comm_ring : comm_ring (⨁ i, A i) := {
 
 end comm_ring
 
+
+/-! ### Instances for `A 0`
+
+The various `g*` instances are enough to promote the `add_comm_monoid (A 0)` structure to various
+types of multiplicative structure.
+-/
+
+section grade_zero
+
+section one
+variables [has_zero ι] [ghas_one A] [Π i, add_comm_monoid (A i)]
+
+/-- `1 : A 0` is the value provided in `direct_sum.ghas_one.one`. -/
+@[nolint unused_arguments]
+instance grade_zero.has_one : has_one (A 0) :=
+⟨ghas_one.one⟩
+
+lemma of_zero_one : of _ 0 (1 : A 0) = 1 := rfl
+
+end one
+
+section mul
+variables [add_monoid ι] [Π i, add_comm_monoid (A i)] [ghas_mul A]
+
+/-- `(•) : A 0 → A i → A i` is the value provided in `direct_sum.ghas_mul.mul`, composed with
+an `eq.rec` to turn `A (0 + i)` into `A i`.
+-/
+instance grade_zero.has_scalar (i : ι) : has_scalar (A 0) (A i) :=
+{ smul := λ x y, (zero_add i).rec (ghas_mul.mul x y)}
```

Comment:
```suggestion
{ smul := λ x y, (zero_add i).rec (ghas_mul.mul x y) }
```
Very minor spacing issue (same goes for `grade_zero.has_mul` and `of_zero_ring_hom` below).
Alternatively, `\< ... \>` might look cleaner?